### ISSUE TITLE
KSCrash crashes when attempting to parse a report it created

### ISSUE THREAD

**Issue by alanzeino** [kstenerud/KSCrash#238]
In `1.15.8`, `-[KSCrashReportFilterAppleFmt headerStringForReport:]` crashes when attempting to convert a report that KSCrash itself created. At some point in the past KSCrash stopped writing string dates and started writing timestamp dates instead.

But this code, still assumes they're string dates:

```
- (NSString*) headerStringForReport:(NSDictionary*) report
{
    NSDictionary* system = [self systemReport:report];
    NSDictionary* reportInfo = [self infoReport:report];
    NSString *reportID = [reportInfo objectForKey:@KSCrashField_ID];
    NSDate* crashTime = [g_rfc3339DateFormatter dateFromString:[reportInfo objectForKey:@KSCrashField_Timestamp]];
```

`dateFromString:` assumes it's getting a string, but it's actually getting an NSNumber.

We got the report by doing this:

```
    int reportCount = kscrs_getReportCount();
    int64_t *reportsArray = malloc(reportCount * sizeof(int64_t));
    int foundReportsCount = kscrs_getReportIDs(reportsArray, reportCount);

    NSMutableArray<NSDictionary *> *convertedReports = [[NSMutableArray alloc] initWithCapacity:reportCount];
    for (int i = 0; i < reportCount; i++) {
        int64_t reportID = reportsArray[i];
        char *report = kscrs_readReport(reportID);
        NSString *reportString = [[NSString alloc] initWithUTF8String:report];
        NSData *data = [reportString dataUsingEncoding:NSUTF8StringEncoding];
        NSError *jsonError = nil;
        NSDictionary *reportDictionary = [NSJSONSerialization JSONObjectWithData:data options:(NSJSONReadingOptions)0 error:&jsonError];
        if (reportDictionary) {
            [convertedReports addObject:reportDictionary];
        } else {
            NSLog(@"Unable to convert crash report to Dictionary type. Error: %@", jsonError.localizedDescription);
        }
        free(report);
    }
    free(reportsArray);
```

Those reports in `convertedReports` have `report.timestamp` as NSNumber, but even the raw JSON is still a integer timestamp:

```
    "report": {
        "version": "3.2.0",
        "id": "01575AAA-CDF3-42F5-B337-FDD6C886FD9E",
        "process_name": "UberEats",
        "timestamp": 1495829821,
        "type": "standard"
    },
```

This is confirmed by this line of code:

```
        writer->addIntegerElement(writer, KSCrashField_Timestamp, time(NULL));
```

In short `KSCrashField_Timestamp` is written one way, but expected to be parsed another (the old way).

**Comment by naftaly** [comment]
Report parsing crashes have been fixed in subsequent versions.
